pkg/plugins/golang/v2/api.go

```diff
@@ -44,6 +45,8 @@ type createAPISubcommand struct {
 
 	options *Options
 
+	resource resource.Resource
```

Comment:
This field isn't set and passed to the scaffold constructor until `Run()` and no flags use it unless I'm mistaken, so what's the point of having a field instead of a local variable within `Run()`? The `resource.Validate()` call doesn't seem to be doing anything.

Reply:
I'm not sure what do you mean here. Do you mean that `resource.Validate()` doesn't change it? That we can have a local variable in `Run` and pass it to `Validate` and `GetScaffolder` methods? In this case `Validate` and `GetScaffolder` methods are used for all plugins through `cmdutil.Run` and not all of them have a `resource.Resource`.

Reply:
I think I see what you mean. I forgot that `cmdutil.Run()` will call `Validate()` before `GetScaffolder()` and therefore `p.resource` is set up before `Validate()` is called.